### PR TITLE
[FIX] [8.0] [i178] Improved reporting BR

### PR DESCRIPTION
**Pull request by sudhir-serpentcs**
@elicoidal  @seb-elico @victormartinelicocorp 
Improved report.

### CODE REVIEW

File: business_requirement_deliverable/report/br_deliverable_sale_report.py [hunk 1]
```diff
@@ -10,19 +10,28 @@ class BusinessRequirementDeliverableSaleReport(models.Model):
...
+    br_name = fields.Char('Bus. Requirement', readonly=True)
+    responsible_id = fields.Many2one('res.users', 'Responsible', readonly=True)
     partner_id = fields.Many2one('res.partner', 'Customer', readonly=True)
     project_id = fields.Many2one('project.project', 'Master Project',
                                  readonly=True)
-    change_request = fields.Boolean('Change Request?', readonly=True)
     priority = fields.Selection([('0', 'Low'), ('1', 'Normal'), ('2', 'High')],
                                 'Priority', readonly=True)
+    state = fields.Selection(
+        [('draft', 'Draft'), ('confirmed', 'Confirmed'),
```
Comment (elicoidal) [inline business_requirement_deliverable/report/br_deliverable_sale_report.py]: better one value per line

**Comment by elicoidal** [comment]
#178

**Review by elicoidal — CHANGES_REQUESTED** [review]
mild details.
I do not see the new menu though nor the modification on the dimension sorting

File: business_requirement_deliverable_cost/report/br_deliverable_cost_report.py [hunk 2]
```diff
@@ -10,26 +10,35 @@ class BusinessRequirementDeliverableCostReport(models.Model):
...
     project_id = fields.Many2one('project.project',
                                  'Master Project',
                                  readonly=True)
-    change_request = fields.Boolean('Change Request?', readonly=True)
     priority = fields.Selection([('0', 'Low'), ('1', 'Normal'),
                                  ('2', 'High')],
                                 'Priority',
                                 readonly=True)
+    state = fields.Selection(
+        [('draft', 'Draft'), ('confirmed', 'Confirmed'),
```
Comment (elicoidal) [inline business_requirement_deliverable_cost/report/br_deliverable_cost_report.py]: One selection per line to improve readibility

**Review by elicoidal — CHANGES_REQUESTED** [review]
Is sorting of fields in the report view fixed?

**Comment by elicoidal** [comment]
@sudhir-serpentcs retested but basically same as before. Besides, it seems that I might have found a calculation bug.
Here is my BR setup in runbot:
screenshot from 2017-07-10 15-09-35

But here is the result in the reports (only one line appears):
screenshot from 2017-07-10 15-07-33
screenshot from 2017-07-10 15-07-51

**Comment by elicoidal** [comment]
@sudhir-serpentcs 
New tests failing:
screenshot from 2017-07-13 17-00-25
screenshot from 2017-07-13 17-00-08

**Comment by elicoidal** [comment]
@sudhir-serpentcs Can you recheck everthing because this is still failing basic tests (bases on your data in runbot):

screenshot from 2017-07-17 14-52-45
screenshot from 2017-07-17 14-52-38
screenshot from 2017-07-17 14-53-05

**Comment by sudhir-serpentcs** [comment]
@elicoidal Could you please review.

**Comment by elicoidal** [comment]
I retested but it still fails: maybe my requirement is not clear enough.
screenshot from 2017-07-19 16-56-40
screenshot from 2017-07-19 16-56-11
screenshot from 2017-07-19 16-56-04

Let me know if you need a call

**Comment by sudhir-serpentcs** [comment]
@elicoidal I think we need to discuss on call.

**Comment by sudhir-serpentcs** [comment]
@elicoidal Please test the report and review.

As per our discussion on call, you gave following calculation:

**Total Revenue = DL Sales Price * DL Qty
Sales Price = Total Revenue / DL Qty**
Here sales price and total revenue will be same.

---
Same goes for Avg Price and Total Cost:
**Total Cost = RL Price * RL Qty
Avg Price = Total Cost / RL Qty**
Here as well total cost and avg price will be same.

Let me know if you want me to change the calculation or let it be as per our discussion
dl_cost_analysis

**Comment by elicoidal** [comment]
@sudhir-serpentcs I have had a lot of reflexion about the reporting topic and I will close this PR for the following reasons:
* We still cannot stabilize it
* But most important, this is somehow duplicated with the pivot view in BRDL and BRRL

I prefer to improve the other 2 views accordingly forget about this one.

**Pull request closed without merging**